src/compiler/types.ts

```diff
@@ -4099,6 +4099,7 @@ namespace ts {
         /** configFile is set as non enumerable property so as to avoid checking of json source files */
         /* @internal */ readonly configFile?: JsonSourceFile;
         declaration?: boolean;
+        /* @internal */ disableLanguageService?: boolean;

```

Comment:
i would have one command-line option indicating the desire to have the server working in syntactic mode only, and under the covers, use that to set `disableLanguageService` to true.